pkg/apis/stork/v1alpha1/backuplocation.go

```diff
@@ -0,0 +1,179 @@
+package v1alpha1
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// BackupLocationResourceName is name for "backuplocation" resource
+	BackupLocationResourceName = "backuplocation"
+	// BackupLocationResourcePlural is plural for "backuplocation" resource
+	BackupLocationResourcePlural = "backuplocations"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupLocation represents a backuplocation object
+type BackupLocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Location          BackupLocationItem `json:"location"`
+}
+
+// BackupLocationItem is the spec used to store a backup location
+// Only one of S3Config, AzureConfig or GoogleConfig should be specified and
+// should match the Type field. Members of the config can be specified inline or
+// through the SecretConfig
+type BackupLocationItem struct {
+	Type BackupLocationType `json:"type"`
+	// Path is either the bucket or any other path for the backup location
+	Path         string        `json:"path"`
+	S3Config     *S3Config     `json:"s3Config"`
```

Comment:
Let's have omitempty, else it will all of them will show when trying to edit.

Reply:
Added omitempty for the s3, azure and google configs